File: PS_160586.kt
class Solution {
    val map: HashMap<Char, Int> = hashMapOf()
    fun solution(keymap: Array<String>, targets: Array<String>): ArrayList<Int> {
        val answer: ArrayList<Int> = arrayListOf()

        keymap.forEach {
            it.forEachIndexed { index, c ->
                if(map.containsKey(c)) {
                    if(map[c]!! > index+1) {
                        map.remove(c)
                        map[c] = index + 1
                    }
                } else {
                    map[c] = index+1
                }
            }
        }

        targets.forEach { s ->
            var temp = 0
            for(c in s) {
                if(map.containsKey(c)) {
                    temp += map[c]!!
                } else {
                    temp = -1
                    break
                }
            }
            answer.add(temp)
        }

        return answer
    }
}

File: PS_160586.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

map<char, int> m;

vector<int> solution(vector<string> keymap, vector<string> targets) {
    vector<int> answer;
    
    for(auto &k : keymap) {
        for(int i = 0; i < k.size(); i++) {
            if(m.find(k[i]) != m.end() && m[k[i]] > i+1) {
                m.erase(k[i]);
            }
            m.insert({k[i], i+1});
        }
    }
    
    for(auto &t : targets) {
        int temp = 0;
        for(int i = 0; i < t.size(); i++) {
            if(m.find(t[i]) != m.end()) {
                temp += m[t[i]];   
            } else {
                temp = -1;
                break;
            }
        }
        answer.push_back(temp);
    }
    
    return answer;
}